LocalPayment/src/main/java/com/braintreepayments/api/localpayment/LocalPaymentClient.kt

```diff
@@ -0,0 +1,258 @@
+            .launchAsNewTask(braintreeClient.launchesBrowserSwitchAsNewTask())
+            .url(Uri.parse(localPaymentAuthRequestParams.approvalUrl))
+
+        try {
+            browserSwitchOptions.metadata(
+                JSONObject()
+                    .put(
+                        "merchant-account-id",
+                        localPaymentAuthRequestParams.request.merchantAccountId
+                    )
+                    .put("payment-type", localPaymentAuthRequestParams.request.paymentType)
+                    .put("has-user-location-consent", hasUserLocationConsent)
+            )
+        } catch (e: JSONException) {
+            authRequestFailure(
+                BraintreeException("Error parsing local payment request"),
+                callback
+            )
+            return
+        }
+
+        localPaymentAuthRequestParams.browserSwitchOptions = browserSwitchOptions
+        callback.onLocalPaymentAuthRequest(
+            LocalPaymentAuthRequest.ReadyToLaunch(
+                localPaymentAuthRequestParams
+            )
+        )
+        sendAnalyticsEvent(LocalPaymentAnalytics.BROWSER_SWITCH_SUCCEEDED)
+    }
+
+    private fun authRequestFailure(error: Exception, callback: LocalPaymentAuthCallback) {
+        sendAnalyticsEvent(LocalPaymentAnalytics.PAYMENT_FAILED)
+        callback.onLocalPaymentAuthRequest(LocalPaymentAuthRequest.Failure(error))
+    }
+
+    /**
+     * After receiving a result from the web authentication flow via
+     * [LocalPaymentLauncher.handleReturnToAppFromBrowser], pass the
+     * [LocalPaymentAuthResult.Success] returned to this method to tokenize the local
+     * payment method and receive a [LocalPaymentNonce] on success.
+     *
+     * @param context                         Android Context
+     * @param localPaymentAuthResult a [LocalPaymentAuthResult.Success] received from
+     * [LocalPaymentLauncher.handleReturnToAppFromBrowser]
+     * @param callback                        [LocalPaymentInternalTokenizeCallback]
+     */
+    fun tokenize(
+        context: Context,
+        localPaymentAuthResult: LocalPaymentAuthResult.Success,
+        callback: LocalPaymentTokenizeCallback
+    ) {
+        val browserSwitchResult: BrowserSwitchFinalResult.Success = localPaymentAuthResult
+            .paymentAuthInfo.browserSwitchResultInfo
+
+        val metadata: JSONObject? = browserSwitchResult.requestMetadata
+        val merchantAccountId = Json.optString(metadata, "merchant-account-id", null)
+        val hasUserLocationConsent = Json.optBoolean(metadata, "has-user-location-consent", false)
+
+        val deepLinkUri: Uri = browserSwitchResult.returnUrl
+        if (deepLinkUri == null) {
+            tokenizeFailure(
+                BraintreeException(
+                    "LocalPayment encountered an error, return URL is " +
+                            "invalid."
+                ),
+                callback
+            )
```

Comment:
```suggestion
            val errorMessage = "LocalPayment encountered an error, return URL is invalid."
            tokenizeFailure(BraintreeException(errorMessage), callback)
```
nit pick: I feel like this string could read more cleanly as a one liner.